src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Metadata/Models/ContentPartFieldDefinitionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+using OrchardCore.ContentManagement.Metadata.Settings;
+
+namespace OrchardCore.ContentManagement.Metadata.Models
+{
+    public static class ContentPartFieldDefinitionExtensions
+    {
+        public static bool IsNamedPart(this ContentPartFieldDefinition fieldDefinition)
+            => fieldDefinition.PartDefinition.IsReusable() && fieldDefinition.ContentTypePartDefinition.Name != fieldDefinition.PartDefinition.Name;
+
+        /// <summary>
+        /// Returns reusable field wrapper class name.
+        /// </summary>
+        /// <param name="fieldDefinition">The <see cref="ContentPartFieldDefinition" />.</param>
+        /// <returns></returns>
+        public static string GetReusableFieldWrapperClassName(this ContentPartFieldDefinition fieldDefinition)
```

Comment:
You should not need `GetReusableFieldWrapperClassName` or `GetFieldWrapperClassName` methods. If you do need then, then they do not belong in this class and should have different name. Note they do not return ClassName any more.